include/CL/opencl.hpp

```diff
@@ -10253,6 +10342,530 @@ namespace compatibility {
+                default_error_ = error;
+            }
+            else {
+                default_ = CommandBufferKhr(vector<CommandQueue>{command_queue}, 0, &default_error_);
+            }
+        }
+#if defined(CL_HPP_ENABLE_EXCEPTIONS)
+        catch (cl::Error &e) {
+            default_error_ = e.err();
+        }
+#endif
+    }
+
+    /*! \brief Create the default command buffer khr.
+     *
+     * This sets @c default_. It does not throw
+     * @c cl::Error.
+     */
+    static void makeDefaultProvided(const CommandBufferKhr &c) {
+        default_ = c;
+    }
+
+public:
+#ifdef CL_HPP_UNIT_TEST_ENABLE
+    /*! \brief Reset the default.
+    *
+    * This sets @c default_ to an empty value to support cleanup in
+    * the unit test framework.
+    * This function is not thread safe.
+    */
+    static void unitTestClearDefault() {
+        default_ = CommandBufferKhr();
+    }
+#endif // #ifdef CL_HPP_UNIT_TEST_ENABLE
+
+    static CommandBufferKhr getDefault(cl_int * err = NULL)
+    {
+        std::call_once(default_initialized_, makeDefault);
+        detail::errHandler(default_error_, __CREATE_COMMAND_BUFFER_KHR_ERR);
+        if (err != NULL) {
+            *err = default_error_;
+        }
+        return default_;
+    }
+
+    static CommandBufferKhr setDefault(const CommandBufferKhr &default_cmd_buffer_khr)
+    {
+        std::call_once(default_initialized_, makeDefaultProvided, std::cref(default_cmd_buffer_khr));
+        detail::errHandler(default_error_);
+        return default_;
+    }
+
+    //! \brief Default constructor - initializes to NULL.
+    CommandBufferKhr() : detail::Wrapper<cl_type>() { }
+
+    explicit CommandBufferKhr(const vector<CommandQueue> &queues,
+        cl_command_buffer_properties_khr properties = 0,
+        cl_int* errcode_ret = NULL)
+    {
+        cl_int error = CL_SUCCESS;
+        cl_command_buffer_properties_khr command_buffer_properties[] = {
+            CL_COMMAND_BUFFER_FLAGS_KHR, properties, 0
+        };
+
+        /* initialization of addresses to extension functions */
+        error = initExtensions(queues[0].getInfo<CL_QUEUE_DEVICE>());
+
+        if (error == CL_SUCCESS)
+        {
+            object_ = pfn_clCreateCommandBufferKHR(queues.size(),
+                &queues[0](),
+                command_buffer_properties,
+                &error);
+        }
+
+        detail::errHandler(error, __CREATE_COMMAND_BUFFER_KHR_ERR);
+        if (errcode_ret != NULL) {
+            *errcode_ret = error;
+        }
+    }
+
+    explicit CommandBufferKhr(const cl_command_buffer_khr& commandBufferKhr, bool retainObject = false) :
+        detail::Wrapper<cl_type>(commandBufferKhr, retainObject) { }
+
+    CommandBufferKhr& operator = (const cl_command_buffer_khr& rhs)
+    {
+        detail::Wrapper<cl_type>::operator=(rhs);
+        return *this;
+    }
+
+    /*! \brief Copy constructor to forward copy to the superclass correctly.
+     * Required for MSVC.
```

Comment:
Is this still true?

Reply:
I wasn't sure of that. I am working on Linux OS and I had to ask colleague to verify this on Windows. All tests passed, so I removed this part of code.